File: desktop/src/types/events.ts
export type ContentEvent = {
  id: number;
  name: string;
  event_type: string | null;
  event_date: string | null;
  folder_name: string | null;
  root_path: string | null;
  event_path: string | null;
  status: string;
  metadata_date_source: string | null;
  notes: string | null;
  archived_at: string | null;
  created_at: string;
  updated_at: string;
};

export type EventListResponse = {
  items: ContentEvent[];
};

export type EventCreate = {
  name: string;
  event_type: string | null;
  event_date: string;
  notes: string | null;
};

export type EventUpdate = {
  name?: string;
  event_type?: string | null;
  event_date?: string;
  notes?: string | null;
};

File: backend/app/schemas/events.py
from __future__ import annotations

from datetime import date, datetime

from pydantic import BaseModel, Field


class EventCreate(BaseModel):
    name: str = Field(min_length=1, max_length=180)
    event_type: str | None = Field(default=None, max_length=80)
    event_date: date
    notes: str | None = None


class EventUpdate(BaseModel):
    name: str | None = Field(default=None, min_length=1, max_length=180)
    event_type: str | None = Field(default=None, max_length=80)
    event_date: date | None = None
    notes: str | None = None


class EventResponse(BaseModel):
    id: int
    name: str
    event_type: str | None
    event_date: date | None
    folder_name: str | None
    root_path: str | None
    event_path: str | None
    status: str
    metadata_date_source: str | None
    notes: str | None
    archived_at: datetime | None
    created_at: datetime
    updated_at: datetime


class EventListResponse(BaseModel):
    items: list[EventResponse]
